gateway/router.go

```diff
@@ -49,7 +49,11 @@ func New(p Params) (*gin.Engine, error) {
 				return err
 			}
 			p.Logger.Info("http server listening", "address", srv.Addr)
-			go srv.Serve(ln)
+			go func() {
+				if err := srv.Serve(ln); err != nil {
+					p.Logger.Error("unable to start the server", "err", err)
```

Comment:
Let's use fx `shutdowner` to shutdown the application when this occurs during start up, what do you think?

To get the `shutdowner` add it as a "parameter" to the `Params` struct above like `Shutdowner fx.Shutdowner` and then should be able to call it after the logging line.